src/multilevel-atom.cpp

```diff
@@ -153,13 +159,14 @@ void multilevel_susceptibility::init_internal_data(
      pointer in d to point to the corresponding data in d->data, so
      that we don't have to remember in other functions how d->data is
      laid out. */
-
+  
   d->GammaInv = d->data;
   for (int i = 0; i < L; ++i)
     for (int j = 0; j < L; ++j)
-      d->GammaInv[i*L + j] = (i == j) + Gamma[i*L + j] * dt/2;
+      d->GammaInv[i*L + j] = (i == j) - (Gamma[i*L + j] + Rp[i*L + j]) * dt/2;
+
   if (!invert(d->GammaInv, L)) 
-    abort("multilevel_susceptibility: I + Gamma*dt/2 matrix singular");
+    abort("multilevel_susceptibility: I - (Gamma+R)*dt/2 matrix singular");
```

Comment:
I don't understand your comment; regardless of what the interface looks like, couldn't we just combine R and Γ internally?